### PR TITLE
[DOC] Update useful_tools.md

### PR DESCRIPTION
**Pull request by matrixgame2018**
更新了浏览数据部分的文档

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/user_guides/useful_tools.md [hunk 1]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+```
+
+**所有参数的说明**：
+
+- `config` : 模型配置文件的路径。
+- `-o, --output-dir`: 保存图片文件夹，如果没有指定，默认为 `'./output'`。
+- **`-p, --phase`**: 可视化数据集的阶段，只能为 `['train', 'val', 'test']` 之一，默认为 `'train'`。
+- **`-n, --show-number`**: 可视化样本数量。如果没有指定，默认展示数据集的所有图片。
+- **`-m, --mode`**: 可视化的模式，只能为 `['original', 'transformed','pipeline']` 之一。 默认为`'transformed'`。
+- `--cfg-options` : 对配置文件的修改，参考[[学习配置文件](https://github.com/open-mmlab/mmyolo/compare/main...matrixgame2018:mmyolo-1:update-tools/config.md)](./config.md)。
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:440]:
```suggestion
- `--cfg-options` : 对配置文件的修改，参考[学习配置文件](./config.md)。
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 2]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+
+```
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:442]:
```suggestion
```{note}
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 3]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
 ```shell
 python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --area-size small
 ```
+## 浏览数据集
+
+```
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:422]:
```suggestion
```shell
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 4]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+- 如果 `--mode` 设置为 'original'，则获取原始图片；
+- 如果 `--mode` 设置为 'transformed'，则获取预处理后的图片；
+- 如果 `--mode` 设置为 'pipeline'，则获得数据流水线所有中间过程图片。
+```
+
+**示例**：
+
+1. **'original'** 模式 ：
+
+```
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:453]:
```suggestion
```shell
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 5]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+
+- `--phase val`: 可视化验证集, 可简化为 `-p val`;
+- `--output-dir tmp`: 可视化结果保存在 "tmp" 文件夹, 可简化为 `-o tmp`;
+- `--mode original`: 可视化原图, 可简化为 `-m original`;
+- `--show-number 100`: 可视化100张图，可简化为 `-n 100`;
+
+
+2.**'transformed'** 模式 ：
+
+```
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:465]:
```suggestion
```shell
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 6]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+python ./tools/analysis_tools/browse_dataset.py configs/yolov5/yolov5_balloon.py 
+```
+
+3.**'pipeline'** 模式 ：
+
+```
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:471]:
```suggestion
```shell
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 7]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
...
+    [--cfg-options ${CFG_OPTIONS}]
+```
+
+**所有参数的说明**：
+
+- `config` : 模型配置文件的路径。
+- `-o, --output-dir`: 保存图片文件夹，如果没有指定，默认为 `'./output'`。
+- **`-p, --phase`**: 可视化数据集的阶段，只能为 `['train', 'val', 'test']` 之一，默认为 `'train'`。
+- **`-n, --show-number`**: 可视化样本数量。如果没有指定，默认展示数据集的所有图片。
+- **`-m, --mode`**: 可视化的模式，只能为 `['original', 'transformed','pipeline']` 之一。 默认为`'transformed'`。
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:439]:
```suggestion
- **`-m, --mode`**: 可视化的模式，只能为 `['original', 'transformed', 'pipeline']` 之一。 默认为 `'transformed'`。
```

**Comment by RangeKing** [comment]
@matrixgame2018, thanks for your contribution. Please sign the CLA and fix the lint issue.

File: docs/zh_cn/user_guides/useful_tools.md [hunk 8]
```diff
@@ -417,3 +417,59 @@ python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --classes cat dog person
 ```shell
 python tools/misc/extract_subcoco.py ${ROOT} ${OUT_DIR} --area-size small
 ```
+## 浏览数据集
```
Comment (RangeKing) [inline docs/zh_cn/user_guides/useful_tools.md:420]: 建议放到 ##可视化 ###可视化数据集 下

**Comment by hhaAndroid** [comment]
@matrixgame2018 please fix the conficts

File: docs/zh_cn/user_guides/useful_tools.md [hunk 9]
```diff
@@ -75,47 +75,60 @@ python tools/analysis_tools/browse_coco_json.py --data-root './data/coco' \
...
 ### 可视化数据集
 
-脚本 `tools/analysis_tools/browse_dataset.py` 能够帮助用户去直接窗口可视化 config 配置中数据处理部分，同时可以选择保存可视化图片到指定文件夹内。
-
-```shell
-python tools/analysis_tools/browse_dataset.py ${CONFIG} \
-                                              [--out-dir ${OUT_DIR}] \
-                                              [--not-show] \
-                                              [--show-interval ${SHOW_INTERVAL}]
+```
```
Comment (PeterH0323) [inline docs/zh_cn/user_guides/useful_tools.md:78]:
```suggestion
```shell
```

File: docs/zh_cn/user_guides/useful_tools.md [hunk 10]
```diff
@@ -75,47 +75,60 @@ python tools/analysis_tools/browse_coco_json.py --data-root './data/coco' \
...
+3.**'pipeline'** 模式 ：
 
 ```shell
-python tools/analysis_tools/browse_dataset.py 'configs/yolov5/yolov5_s-v61_syncbn_8xb16-300e_coco.py' \
-                                              --out-dir 'work_dirs/browse_dataset' \
-                                              --not-show
+python ./tools/analysis_tools/browse_dataset.py configs/yolov5/yolov5_balloon.py -m pipeline
 ```
 
+![9041866c28ffef58c2597bc8cf6c272](https://user-images.githubusercontent.com/45811724/204810831-0fbc7f1c-0951-4be1-a11c-491cf0d194f6.png)
```
Comment (PeterH0323) [inline docs/zh_cn/user_guides/useful_tools.md:130]:
```suggestion
<div align=center>
<img src="https://user-images.githubusercontent.com/45811724/204810831-0fbc7f1c-0951-4be1-a11c-491cf0d194f6.png" alt="Image">
</div>
```

**Review by PeterH0323 — APPROVED** [review]
LGTM

**Pull request merged**